multiapps-common/src/test/java/org/cloudfoundry/multiapps/common/util/MapUtilTest.java

```diff
@@ -36,4 +40,91 @@ void testParseBooleanFlag() {
         assertThrows(ContentException.class, () -> MapUtil.parseBooleanFlag(TEST_PARAMETERS, "incorrectTypeFlag1", true));
         assertThrows(ContentException.class, () -> MapUtil.parseBooleanFlag(TEST_PARAMETERS, "incorrectTypeFlag2", true));
     }
+
+    @Test
+    void testCast_withNullMap() {
+        Map<Object, Object> result = MapUtil.cast(null);
+
+        assertNull(result);
+    }
+
+    @Test
+    void testCast_withEmptyMap() {
+        Map<Object, Object> emptyMap = Collections.emptyMap();
+
+        Map<Object, Object> result = MapUtil.cast(emptyMap);
+
+        assertNotNull(result);
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    void testAddNonNull_withNonNullValue() {
+        Map<String, String> map = new HashMap<>();
+        String key = "key1";
+        String value = "value1";
+
+        MapUtil.addNonNull(map, key, value);
+
+        assertEquals(1, map.size());
+        assertEquals(value, map.get(key));
+    }
+
+    @Test
+    void testAddNonNull_withNullValue() {
+        Map<String, String> map = new HashMap<>();
+        String key = "key1";
+        String value = null;
+
+        MapUtil.addNonNull(map, key, value);
+
+        assertTrue(map.isEmpty());
+    }
+
+    @Test
+    void testMergeSafely_withBothMapsNonNull() {
```

Comment:
maybe add a the same key to both maps to check if it's overwritten correctly in this test.